src/main/java/hunternif/mc/atlas/client/gui/core/GuiComponent.java

```diff
@@ -438,69 +441,90 @@ protected boolean isMouseInRadius(int x, int y, int radius) {
-			int i2 = 1347420415;
-			int j2 = (i2 & 16711422) >> 1 | i2 & -16777216;
-			this.drawGradientRect(i1 - 3, j1 - 3 + 1, i1 - 3 + 1, j1 + k1 + 3 - 1, i2, j2);
-			this.drawGradientRect(i1 + k + 2, j1 - 3 + 1, i1 + k + 3, j1 + k1 + 3 - 1, i2, j2);
-			this.drawGradientRect(i1 - 3, j1 - 3, i1 + k + 3, j1 - 3 + 1, i2, i2);
-			this.drawGradientRect(i1 - 3, j1 + k1 + 2, i1 + k + 3, j1 + k1 + 3, j2, j2);
-
-			for (int k2 = 0; k2 < lines.size(); ++k2) {
-				String s1 = lines.get(k2);
-				font.drawStringWithShadow(s1, i1, j1, -1);
-
-				if (k2 == 0) {
-					j1 += 2;
+
+
+			this.zLevel = 300;
+			final int backgroundColor = 0xF0100010;
+			this.drawGradientRect(tooltipX - 3, tooltipY - 4, tooltipX + tooltipTextWidth + 3, tooltipY - 3, backgroundColor, backgroundColor);
+			this.drawGradientRect(tooltipX - 3, tooltipY + tooltipHeight + 3, tooltipX + tooltipTextWidth + 3, tooltipY + tooltipHeight + 4, backgroundColor, backgroundColor);
+			this.drawGradientRect(tooltipX - 3, tooltipY - 3, tooltipX + tooltipTextWidth + 3, tooltipY + tooltipHeight + 3, backgroundColor, backgroundColor);
+			this.drawGradientRect(tooltipX - 4, tooltipY - 3, tooltipX - 3, tooltipY + tooltipHeight + 3, backgroundColor, backgroundColor);
+			this.drawGradientRect(tooltipX + tooltipTextWidth + 3, tooltipY - 3, tooltipX + tooltipTextWidth + 4, tooltipY + tooltipHeight + 3, backgroundColor, backgroundColor);
+			final int borderColorStart = 0x505000FF;
+			final int borderColorEnd = (borderColorStart & 0xFEFEFE) >> 1 | borderColorStart & 0xFF000000;
+			this.drawGradientRect(tooltipX - 3, tooltipY - 3 + 1, tooltipX - 3 + 1, tooltipY + tooltipHeight + 3 - 1, borderColorStart, borderColorEnd);
+			this.drawGradientRect(tooltipX + tooltipTextWidth + 2, tooltipY - 3 + 1, tooltipX + tooltipTextWidth + 3, tooltipY + tooltipHeight + 3 - 1, borderColorStart, borderColorEnd);
+			this.drawGradientRect(tooltipX - 3, tooltipY - 3, tooltipX + tooltipTextWidth + 3, tooltipY - 3 + 1, borderColorStart, borderColorStart);
+			this.drawGradientRect(tooltipX - 3, tooltipY + tooltipHeight + 2, tooltipX + tooltipTextWidth + 3, tooltipY + tooltipHeight + 3, borderColorEnd, borderColorEnd);
+
+			MinecraftForge.EVENT_BUS.post(new RenderTooltipEvent.PostBackground(ItemStack.EMPTY, textLines, tooltipX, tooltipY, font, tooltipTextWidth, tooltipHeight));
+			int tooltipTop = tooltipY;
+
+			for (int lineNumber = 0; lineNumber < textLines.size(); ++lineNumber) {
+				String s1 = textLines.get(lineNumber);
+				font.drawStringWithShadow(s1, (float)tooltipX, (float)tooltipY, 0xFFFFFFFF);
+
+				if (lineNumber == 0) {
+					tooltipY += 2;
 				}
 
-				j1 += 10;
+				tooltipY += 10;
 			}
+
+			MinecraftForge.EVENT_BUS.post(new RenderTooltipEvent.PostText(ItemStack.EMPTY, textLines, tooltipX, tooltipTop, font, tooltipTextWidth, tooltipHeight));
+
 			if (stencilEnabled) GL11.glEnable(GL11.GL_STENCIL_TEST);
+			GlStateManager.enableLighting();

```

Comment:
I'm not too familiar with OpenGL, but I'd expect the exact reverse ordering of these calls when compared to the disable calls at the beginning of this function?